src/python/pants/engine/internals/graph.py

```diff
@@ -818,6 +878,52 @@ async def resolve_dependencies(
     return Addresses(sorted(result))
 
 
+@rule(desc="Resolve direct dependencies")
+async def resolve_dependencies_lite(
+    request: DependenciesRequestLite,
+    union_membership: UnionMembership,
+    registered_target_types: RegisteredTargetTypes,
+    global_options: GlobalOptions,
+) -> Addresses:
+    provided = parse_dependencies_field(
+        request.field,
+        subproject_roots=global_options.options.subproject_roots,
+        registered_target_types=registered_target_types.types,
+        union_membership=union_membership,
+    )
+    literal_addresses = await MultiGet(Get(Address, AddressInput, ai) for ai in provided.addresses)
+    ignored_addresses = set(
+        await MultiGet(Get(Address, AddressInput, ai) for ai in provided.ignored_addresses)
+    )
+
+    # Inject any dependencies.
+    inject_request_types = union_membership.get(InjectDependenciesRequest)
+    injected = await MultiGet(
+        Get(InjectedDependencies, InjectDependenciesRequest, inject_request_type(request.field))
+        for inject_request_type in inject_request_types
+        if isinstance(request.field, inject_request_type.inject_for)
+    )
+
+    # Inject dependencies on all the base target's generated subtargets.
+    subtargets = await Get(
+        Subtargets, Address, request.field.address.maybe_convert_to_base_target()
+    )
```

Comment:
Normally we only do this if inference isn't at play. Because we never use inference here, we always perform this step.